ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationContent.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { CSSProperties, ReactElement, ReactNode } from 'react';
+import { Flex, Label } from '@patternfly/react-core';
+
+type NavigationContentVariant = 'Deprecated' | 'TechPreview';
+
+const badgeMap: Record<NavigationContentVariant, ReactElement> = {
+    Deprecated: (
+        <Label
+            isCompact
+            style={
+                {
+                    '--pf-c-label--BackgroundColor': 'var(--pf-global--disabled-color--200)',
+                } as CSSProperties
+            }
+        >
+            Deprecated
+        </Label>
+    ),
+    TechPreview: (
+        <Label isCompact color="orange">
+            Tech preview
+        </Label>
+    ),
+} as const;
```

Comment:
Curious why we need to make this a constant type, when we are not exporting it, and there is only one usage later in this file?

Reply:
Super question! It might be part of an idiom to derive a type of the keys of the object map, although they are defined the other way around here:

```ts
type NavigationContentVariant = 'Deprecated' | 'TechPreview';

const badgeMap: Record<NavigationContentVariant, ReactElement> = {
```

@dvail Can you help us interpret `as const` either as written or as might be rewritten?

Reply:
> I'm not sure what you mean

Sorry I was unclear. I mean restrict properties **added later** to have values of consistent type.